src/main/duke/Duke.java

```diff
@@ -0,0 +1,42 @@
+package main.duke;
+
+import main.duke.commands.Command;
+import main.duke.io.Parser;
+import main.duke.io.Storage;
+
+import java.util.Scanner;
+
+public class Duke {
+    private static final String WELCOME_MESSAGE = "Hello! I'm Duke \n" + "What can I do for you";
+
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+    private Parser parser;
+
+    public Duke(String dirname, String filename) {
+        this.storage = new Storage(dirname, filename);
+        this.taskList = new TaskList();
+        this.ui =  new Ui();
+        this.parser = new Parser();
+    }
+
+    public void run() throws DukeException{
```

Comment:
I think it should be a white space before the open curly bracket. I noticed the same issue in several other places too.